tests/cases/conformance/types/unknown/unknownControlFlow.ts

```diff
@@ -429,3 +429,25 @@ type AB = "A" | "B";
 function x<T_AB extends AB>(x: T_AB & undefined, y: any) {
     let r2: never = y as T_AB & undefined;
 } 
+
+// Repro from #51538
+
+type Left = 'left';
+type Right = 'right' & { right: 'right' };
+type Either = Left | Right;
+
+function assertNever(v: never): never {
+    throw new Error('never');
+}
+
+function fx20(value: Either) {

```

Comment:
I feel like I'm having deja vu and I've asked this question, but why is this test in a file called `unknownControlFlow.ts`?

Reply:
You did indeed: https://github.com/microsoft/TypeScript/pull/51076#discussion_r998620628. The test is in that file for the same reason.